bundles/org.openhab.binding.meteofrance/src/main/java/org/openhab/binding/meteofrance/internal/handler/VigilanceHandler.java

```diff
@@ -0,0 +1,189 @@
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Wrong department: %s".formatted((config.department)));
+            return;
+        }
+        logger.debug("config department= {}", config.department);
+
+        updateStatus(ThingStatus.UNKNOWN);
+        refreshJob = Optional
+                .of(scheduler.scheduleWithFixedDelay(this::updateAndPublish, 5, config.refresh * 60, TimeUnit.SECONDS));
+    }
+
+    private void disposeJob() {
+        refreshJob.ifPresent(job -> job.cancel(true));
+        refreshJob = Optional.empty();
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the Vigilance Météo handler.");
+        disposeJob();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateAndPublish();
+        }
+    }
+
+    private void updateAndPublish() {
+        getBridgeHandler().ifPresentOrElse(handler -> {
+            Period period = handler.requestPeriod(Term.TODAY);
+            if (period != null) {
+                updateDate(OBSERVATION_TIME, period.beginValidityTime());
+                updateDate(END_TIME, period.endValidityTime());
+            }
+            DomainId domainId = handler.requestMapData(domain, Term.TODAY);
+            ZonedDateTime now = ZonedDateTime.now().withZoneSameInstant(zoneId);
+            if (domainId != null) {
+                Map<Hazard, Risk> channels = new HashMap<>(Hazard.values().length);
+                Hazard.AS_SET.stream().filter(Hazard::isChannel).forEach(hazard -> channels.put(hazard, Risk.VERT));
+
+                domainId.phenomenonItems().forEach(phenomenon -> {
+                    Risk risk = phenomenon.phenomenonMaxColorId();
+                    for (TimelapsItem item : phenomenon.timelapsItems()) {
+                        if (item.contains(now)) {
+                            risk = item.colorId();
+                        }
+                    }
+                    channels.put(phenomenon.phenomenonId(), risk);
+                });
+
+                channels.forEach((k, v) -> updateAlert(k.channelName, v));
+
+                TextBlocItem bloc = handler.requestTextData(domain);
+                if (bloc != null) {
+                    String comment = bloc.blocItems().stream().filter(bi -> bi.type().equals(BlocType.SITUATION))
+                            .flatMap(bi -> bi.textItems().stream()).flatMap(ti -> ti.termItems().stream())
+                            .map(term -> term.getText(now)).collect(Collectors.joining("."));
+                    updateState(COMMENT, comment.isEmpty() ? UnDefType.NULL : StringType.valueOf(comment));
+                }
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "No data available for the department");
```

Comment:
Just a check, what would be the cause of no data being available?

Reply:
I guess that I has happened during tests or developments if it is there.